src/main/java/io/vertx/ext/mail/impl/SMTPConnection.java

```diff
@@ -137,30 +137,14 @@ void writeLine(String str, boolean mayLog) {
       log.debug(str);
     }
     ns.write(str + "\r\n");
-  }
-
-  // write single line not expecting a reply, using drain handler
-  void writeLineWithDrainHandler(String str, boolean mayLog, Handler<Void> handler) {
-    if (mayLog) {
-      log.debug(str);
-    }
     if (ns.writeQueueFull()) {
+      ns.pause();
       ns.drainHandler(v -> {
-        // avoid getting confused by being called twice
-        ns.drainHandler(null);
-        ns.write(str + "\r\n");
-        handler.handle(null);
+        ns.resume();
```

Comment:
why is this needed ? this will resume reading from the same netsocket

Reply:
This must be misunderstanding of how the Streams are used. :)